File: Lintcode 044 -  Minimum Subarray.java
Lintcode 44 -  Minimum Subarray.java

public class Solution {
    /**
     * @param nums: a list of integers
     * @return: A integer indicate the sum of minimum subarray
     */
    public int minSubArray(ArrayList<Integer> nums) {
        // write your code
        if(nums == null || nums.size() == 0) {
            return 0;
        }
        
        int min = Integer.MAX_VALUE;
        int sum = 0;
        for(int i = 0; i< nums.size(); i++) {
            sum += nums.get(i);
            min = Math.min(min, sum);
            sum = Math.min(sum, 0);
        }
        return min;
    }
}

File: Lintcode 044 - Minimum Subarray.cs
using System;
using System.Collections;

namespace CSharp
{
    class Program{
        static void Main()
        {
            int[] A = new int[]{-2, 2, -3, 4, -1, 2, 1, -5, 3};

            int result = MinSub(A);
            Console.WriteLine(result);

        }

        static int MinSub(int[] A){
            int max = int.MaxValue;
            int sum = 0;

            for(int i = 0; i < A.Length; i++)
            {
                sum += A[i];
                max = Math.Min(max, sum);
                sum = Math.Min(0, sum);
            }

            return max;
        }
    }
}
